core/src/main/java/org/pentaho/di/core/plugins/PluginRegistry.java

```diff
@@ -1063,4 +1064,26 @@ public void reset() {
       lock.writeLock().unlock();
     }
   }
+
+  public PluginInterface findPluginWithId( Class<? extends PluginTypeInterface> pluginType, String pluginId, boolean waitForPluginToBeAvailable ) {
+    return waitForPluginToBeAvailable && findPluginWithId(  pluginType,  pluginId ) == null
+      ? waitForPluginToBeAvailable( pluginType,  pluginId, waitForPluginToBeAvailableLimit )
+      : findPluginWithId(  pluginType,  pluginId );
+  }
+
+  public PluginInterface waitForPluginToBeAvailable( Class<? extends PluginTypeInterface> pluginType, String pluginId, int waitLimit ) {
+    int timeToSleep = 1;
```

Comment:
timeToSleep of 1 seems aggressive- for a missing plugin we'll end up calling findPluginWithId 2000 times.  Maybe every 50 or 100 milliseconds?